app/src/main/java/com/mapbox/navigation/examples/basics/ShowCameraTransitionsActivity.kt

```diff
@@ -36,48 +88,317 @@ class ShowCameraTransitionsActivity : AppCompatActivity() {
+        )
+        navigationCamera.registerNavigationCameraStateChangeObserver { navigationCameraState ->
+            // shows/hide the recenter button depending on the camera state
+            when (navigationCameraState) {
+                NavigationCameraState.TRANSITION_TO_FOLLOWING,
+                NavigationCameraState.FOLLOWING ->
+                    binding.recenterButton.visibility = View.INVISIBLE
+                NavigationCameraState.TRANSITION_TO_OVERVIEW,
+                NavigationCameraState.OVERVIEW,
+                NavigationCameraState.IDLE -> binding.recenterButton.visibility = View.VISIBLE
+            }
+        }
+        // set the padding values depending to correctly frame maneuvers and the puck
+        viewportDataSource.overviewPadding = overviewPadding
+        viewportDataSource.followingPadding = followingPadding
+
+        // initialize route line, the withRouteLineBelowLayerId is specified to place
+        // the route line below road labels layer on the map
+        // the value of this option will depend on the style that you are using
+        // and under which layer the route line should be placed on the map layers stack
+        val mapboxRouteLineOptions = MapboxRouteLineOptions.Builder(this)
+            .withRouteLineBelowLayerId("road-label")
+            .build()
+        routeLineApi = MapboxRouteLineApi(mapboxRouteLineOptions)
+        routeLineView = MapboxRouteLineView(mapboxRouteLineOptions)
+
+        // add click listeners for buttons
+        binding.recenterButton.setOnClickListener {
+            navigationCamera.requestNavigationCameraToFollowing()
+        }
+        binding.overviewButton.setOnClickListener {
+            navigationCamera.requestNavigationCameraToOverview()
+        }
+        binding.simulationButton.setOnClickListener {
+            // restarts the simulation
+            startSimulation(route)
+        }
+
+        // start the trip session to being receiving location updates in free drive
+        // and later when a route is set also receiving route progress updates
+        mapboxNavigation.startTripSession()
+
+        // load map style
+        mapboxMap.loadStyleUri(
+            Style.MAPBOX_STREETS
+        ) {
+            // only once the style is loaded expose an ability to add and draw a route
+            binding.routeButton.setOnClickListener {
+                if (mapboxNavigation.getRoutes().isEmpty()) {
+                    // disable navigation camera
+                    navigationCamera.requestNavigationCameraToIdle()
+                    // set a route to receive route progress updates and provide a route reference
+                    // to the viewport data source (via RoutesObserver)
+                    mapboxNavigation.setRoutes(listOf(route))
+                    // enable the camera back
+                    navigationCamera.requestNavigationCameraToOverview()
+
+                    binding.routeButton.text = "clear route"
```

Comment:
Can we extract `clear route` to `strings.xml`? I think we would definitely want to provide translations to this repository and then this wouldn't work.

Reply:
Would we like to translate this repository? The main audience are developers who won't need the translations as they will be interacting with the codebase directly anyway. For the sake of keeping the example self-contained, I wouldn't mind leaving the strings hardcoded.

Reply:
That's a good point. Makes sense, but on the other hand am thinking how does it work for people in EU or other non english speaking geometries. This is not a blocker for this PR, but something we should discuss.